frontend/src/toolbar/ToolbarApp.tsx

```diff
@@ -1,42 +1,41 @@
-import React, { useRef } from 'react'
+import React, { useRef, useState } from 'react'
 import { useSecondRender } from 'lib/hooks/useSecondRender'
 import root from 'react-shadow'
 import { ToolbarContainer } from '~/toolbar/ToolbarContainer'
-import { useMountedLogic } from 'kea'
+import { useValues } from 'kea'
 import { toolbarLogic } from '~/toolbar/toolbarLogic'
 import { ToolbarProps } from '~/types'
 import { Slide, ToastContainer } from 'react-toastify'
 
 type HTMLElementWithShadowRoot = HTMLElement & { shadowRoot: ShadowRoot }
 
 export function ToolbarApp(props: ToolbarProps = {}): JSX.Element {
-    useMountedLogic(toolbarLogic(props))
+    const { jsURL } = useValues(toolbarLogic(props))
 
     const shadowRef = useRef<HTMLElementWithShadowRoot | null>(null)
+    const [didLoadStyles, setDidLoadStyles] = useState(false)
 
     // this runs after the shadow root has been added to the dom
-    const didRender = useSecondRender(() => {
-        function addStyleElementToShadowRoot(element: HTMLElement): void {
-            const shadowRoot =
-                shadowRef.current?.shadowRoot || window.document.getElementById('__POSTHOG_TOOLBAR__')?.shadowRoot
-            shadowRoot?.getElementById('posthog-toolbar-styles')?.appendChild(element)
-        }
-
-        // add styles that webpack has rendered until now
-        const styles = ((window as any)['__PHGTLB_STYLES__'] || []) as unknown as HTMLStyleElement[]
-        if (styles) {
-            styles.forEach((element) => addStyleElementToShadowRoot(element))
-        }
-
-        // add function to render styles in the future
-        ;(window as any)['__PHGTLB_ADD_STYLES__'] = (element: HTMLStyleElement) => addStyleElementToShadowRoot(element)
-    })
+    const didRender = useSecondRender(
+        props.disableExternalStyles
```

Comment:
This `disableExternalStyles` prevents us from blocking and waiting for the stylesheet to open. It's only used inside storybook right now to bypass loading `toolbar.css` into the shadow root. The story lifts the styles into the shadow root in a different way.

Reply:
I tested that I could still see the toolbar's CSS locally with both webpack and esbuild, with both dev server and production builds.